test/renaissance/auctions/auctions_test.exs

```diff
@@ -17,66 +17,79 @@ defmodule Renaissance.Test.AuctionsTest do
     "price" => "10.00"
   }
 
-  def fixture(:user) do
-    user_params = %{email: "test@suite.com", password: "password"}
-    {:ok, user} = Users.register_user(user_params)
-    user
-  end
-
   @auction_two %{
     "title" => "Test Title Two",
     "description" => "Test description two.",
     "end_auction_at" => @valid_end,
     "price" => "15.00"
   }
 
-  describe "auctions" do
-    test "stores a valid auction in the db" do
-      seller_id = fixture(:user).id
-      {:ok, auction_created} = Auctions.create_auction(seller_id, @auction_one)
+  setup _context do
+    user_params = %{email: "test@suite.com", password: "password"}
+    {:ok, user} = Users.register_user(user_params)
+    [user_id: user.id]
+  end
 
-      assert auction_created.title == @auction_one["title"]
-      assert auction_created.description == @auction_one["description"]
-      assert Money.compare(auction_created.price, Money.new(10_00)) == 0
+  describe "create_auction/2" do
+    test "stores a valid auction in the db", %{user_id: seller_id} do
+      {:ok, new_auction} = Auctions.create_auction(seller_id, @auction_one)
+
+      assert new_auction.title == @auction_one["title"]
+      assert new_auction.description == @auction_one["description"]
+      assert Money.compare(new_auction.price, Money.new(10_00)) == 0
     end
 
-    test "does not store when title is blank" do
-      seller_id = fixture(:user).id
+    test "does not store when title is blank", %{user_id: seller_id} do
+      invalid_params = %{@auction_two | "title" => ""}
+      assert {:error, _} = Auctions.create_auction(seller_id, invalid_params)
+      assert Repo.exists?(Auction) == false
+    end
 
-      invalid_params = Map.put(@auction_two, "title", "")
-      Auctions.create_auction(seller_id, invalid_params)
+    test "does not store when invalid seller_id" do
+      {:error, changeset} = Auctions.create_auction(0, @auction_two)
 
-      count = Repo.aggregate(Ecto.Query.from(p in "auctions"), :count, :id)
-      assert 0 == count
+      assert "does not exist" in errors_on(changeset).seller_id
+      assert Repo.exists?(Auction) == false
     end
+  end
 
-    test "does not store when invalid seller_id" do
-      seller_id = fixture(:user).id
-      invalid_seller_id = seller_id * 7 - 1
+  describe "exists?/1" do
+    test "true when auction with given id", %{user_id: seller_id} do
+      {:ok, auction} = Auctions.create_auction(seller_id, @auction_one)
+      assert Auctions.exists?(auction.id) == true
+    end
 
-      exception =
-        assert_raise Ecto.ConstraintError, fn ->
-          Auctions.create_auction(invalid_seller_id, @auction_two)
-        end
+    test "false when no auction with given id" do
+      assert Auctions.exists?(0) == false
```

Comment:
`asset_false` works, no?

Reply:
Resolved with 46d8ca2 & 209badd .

_(Personally, I find `refute` less clear, but that's just me.)_